iMovies/Core/Error/APIError.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  APIError.swift
+//  iMovies
+//
+//  Created by cefalo on 16/1/24.
+//
+
+import Foundation
+
+enum APIError: Error {
+    case invalidResponse
+    case jsonParsingFailure
+    case requestFailed(description: String)
+    case invalidStatusCode(statusCode: Int)
+    case unknownError(error: Error)
+    case unauthorized
+    case custom(error: Error)
+}
+
+
+ 
+
+extension APIError {
+    
+     func mapToMovieError() -> MovieError {
+        switch self {
+        case .invalidResponse:
+            return MovieError.invalidData
```

Comment:
I'm not 100% sure, but you might remove the `MovieError` as it is the return type. Swift compiler should be able to understand if you only write return `.invalidData`